frontend/src/pages/WorkOrderPage.js

```diff
@@ -0,0 +1,240 @@
+            title: 'untitled', 
+            cause: 'no cause', 
+            service_needed: false, 
+            priority: 'MEDIUM', 
+            description: 'no description', 
+            due_date: '2020-11-07T03:54:52.130+00:00',
+            price_estimate: 0
+        }
+    }
+    static navigationOptions = {
+        title: 'Create Work Order',
+    };
+    
+    handleWorkOrder = async() => {
+        try {
+            await createWorkOrder(
+                this.state.property_id,
+                this.state.sector,
+                this.state.type,
+                this.state.title,
+                this.state.cause,
+                this.state.service_needed,
+                this.state.priority,
+                this.state.description,
+                this.state.due_date,
+                this.state.price_estimate).then(async() => {
+                        this.props.navigation.goBack(null)
+                });
+        } catch (err) {
+            console.log(JSON.stringify(err.response))
+            console.log('' + this.state.property_id + '\n' + 
+                this.state.sector + '\n' +
+                this.state.type + '\n' + 
+                this.state.title + '\n' + 
+                this.state.cause + '\n' + 
+                this.state.service_needed + '\n' + 
+                this.state.priority + '\n' + 
+                this.state.description + '\n' + 
+                this.state.due_date + '\n' + 
+                this.state.price_estimate + '\n' +
+                err.response.data);
+        }
+    }
+
+    toggleCorrective = () => {
+        if(this.state.type == 'CM'){
+            this.setState({type: 'IMP'})
+        }
+        else{
+            this.setState({type: 'CM'})
+        }
+
+    }
+
+    togglePreventive = () => {
+        if(this.state.type == 'PM'){
+            this.setState({type: 'IMP'})
+        }
+        else{
+            this.setState({type: 'PM'})
+        }
+
+    }
+
+    handleSector = (sector) => {
+        this.setState({sector: sector})
+    }
+
+    handleTitle = (title) => {
+        this.setState({title: title})
+    }
+    
+    handleCause = (cause) => {
+        this.setState({cause: cause})
+    }
+
+    toggleServiceNeeded = (value) => {
+        this.setState({service_needed: value})
+    }
+
+    handleDescription = (description) => {
+        this.setState({description: description})
+    }
+
+    handlePriority = (priority) => {
+        this.setState({priority: priority})
+    }
+
+    correctiveStyle = function() {
+        if(this.state.type == 'CM'){
+            return workOrderPage.selectedGray;
+        }
+        else{
+            return workOrderPage.unselectedGray;
+        }
+    }
+
+    preventiveStyle = function() {
+        if(this.state.type == 'PM'){
+            return workOrderPage.selectedGray;
+        }
+        else{
+            return workOrderPage.unselectedGray;
+        }
+    }
+    
+    render() {
+        return (
```

Comment:
everything returned here (the view) should be in its own file, called something like WorkOrderComponent, where all the logic and handlers are passed down to that component (see loginPage and loginForm, or SignUpPage and SignUpComponent for an example)